File: docs/manual/java/guide/logging/code/docs/logging/Log4j2Example.java
package docs.logging;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class Log4j2Example {
    private static final Logger LOGGER = LogManager.getLogger();

    public void demonstrateLogging(String msg) {
        LOGGER.debug("Here is a message at the debug level: {}", msg);
    }
}

File: docs/manual/scala/guide/logging/code/docs/scaladsl/logging/Log4j2Example.scala
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.scaladsl.logging

import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

object Log4j2Example {
  private final val Logger: Logger = LogManager.getLogger
}

class Log4j2Example {
  def demonstrateLogging(msg: String): Unit = {
    Log4j2Example.Logger.debug("Here is a message at the debug level: {}", msg)
  }
}
